test/provision/manifest/dns_deployment.yaml

```diff
@@ -140,71 +105,97 @@ spec:
         - containerPort: 10055
           name: metrics
           protocol: TCP
-        volumeMounts:
-        - name: kube-dns-config
-          mountPath: /kube-dns-config
-      - name: dnsmasq
-        image: k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64:1.14.10
-        livenessProbe:
+        readinessProbe:
+          failureThreshold: 3
           httpGet:
-            path: /healthcheck/dnsmasq
-            port: 10054
+            path: /readiness
+            port: 8081
             scheme: HTTP
-          initialDelaySeconds: 60
-          timeoutSeconds: 5
+          initialDelaySeconds: 3
+          periodSeconds: 10
           successThreshold: 1
-          failureThreshold: 5
-        args:
-        - -v=5
+          timeoutSeconds: 5
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        volumeMounts:
+        - mountPath: /kube-dns-config
+          name: kube-dns-config
+      - args:
+        - -v=2
         - -logtostderr
         - -configDir=/etc/k8s/dns/dnsmasq-nanny
         - -restartDnsmasq=true
         - --
         - -k
         - --cache-size=1000
-        - --no-negcache
```

Comment:
We should watch for this, it might be a little dangerous. When negative caching is turned on, it means we'll cache responses saying there is nothing, and it keeps this for some time. This means if something tries to lookup an address too early, it needs to wait a whole TTL to get the address (assuming it came up in the meantime). This won't affect our tests, but do we use this server for to resolve anything? (like the image repository on k8s1?)